packages/workers/src/block-indexer.ts

```diff
@@ -0,0 +1,173 @@
+import { ChainName, type Chain } from '@metamask-institutional/defi-adapters'
+import { ChainIdToChainNameMap } from '@metamask-institutional/defi-adapters/dist/core/constants/chains.js'
+import type { CustomJsonRpcProvider } from '@metamask-institutional/defi-adapters/dist/core/provider/CustomJsonRpcProvider.js'
+
+import { AVERAGE_BLOCKS_PER_10_MINUTES } from '@metamask-institutional/defi-adapters/dist/core/constants/AVERAGE_BLOCKS_PER_10_MINS.js'
+import { AVERAGE_BLOCKS_PER_DAY } from '@metamask-institutional/defi-adapters/dist/core/constants/AVERAGE_BLOCKS_PER_DAY.js'
+import { logger } from './logger.js'
+
+export class BlockRunner {
+  private _provider: CustomJsonRpcProvider
+  private _chainId: Chain
+  private _chainName: string
+
+  private static readonly _BATCH_SIZE = 50
+  private static readonly _MAX_RETRIES = 5
+
+  private _latestBlockNumber: number | undefined
+  private _getStartBlockNumber: () => Promise<number>
+  private _processBlockFn: (blockNumber: number) => Promise<void>
+  private _onError: (latestSafeProcessedBlock: number) => Promise<void>
+
+  constructor({
+    provider,
+    chainId,
+    getStartBlockNumberFn,
+    processBlockFn,
+    onError,
+  }: {
+    provider: CustomJsonRpcProvider
+    chainId: Chain
+    getStartBlockNumberFn: () => Promise<number>
+    processBlockFn: (blockNumber: number) => Promise<void>
+    onError: (latestSafeProcessedBlock: number) => Promise<void>
+  }) {
+    this._chainName = ChainName[chainId]
+    this._processBlockFn = processBlockFn
+    this._getStartBlockNumber = getStartBlockNumberFn
+    this._onError = onError
+
+    this._provider = provider
+    this._chainId = chainId
+    this._chainName = ChainIdToChainNameMap[chainId]
```

Comment:
This is already being setup on line 35.

When exporting `ChainName` from `@metamask-institutional/defi-adapters` directly, it is already exporting the correct one (`ChainIdToChainNameMap`).